src/ReverseProxy/Management/IReverseProxyBuilderExtensions.cs

```diff
@@ -112,4 +118,24 @@ public static IReverseProxyBuilder AddPassiveHealthCheck(this IReverseProxyBuild
         builder.Services.AddSingleton<IPassiveHealthCheckPolicy, TransportFailureRateHealthPolicy>();
         return builder;
     }
+
+    public static IReverseProxyBuilder AddHttpSysDelegation(this IReverseProxyBuilder builder)
+    {
+#if NET6_0_OR_GREATER
+        builder.Services.AddSingleton(p =>
+        {
+            // IServerDelegationFeature isn't added to DI  https://github.com/dotnet/aspnetcore/issues/40043
+            // IServerDelegationFeature may not be set if not http.sys server or the OS doesn't support delegation
+            var delegationFeature = p.GetRequiredService<IServer>().Features?.Get<IServerDelegationFeature>();
+            return new HttpSysDelegationRuleManager(
+                delegationFeature,
+                p.GetRequiredService<ILogger<HttpSysDelegationRuleManager>>());
+        });
+
+        builder.Services.AddSingleton<IHttpSysDelegationRuleManager>(p => p.GetRequiredService<HttpSysDelegationRuleManager>());
```

Comment:
Avoid ordering restrictions if this is intended to be replaced.
```suggestion
        builder.Services.TryAddSingleton<IHttpSysDelegationRuleManager>(p => p.GetRequiredService<HttpSysDelegationRuleManager>());
```

Reply:
Not needed anymore.